doc/mgr/rgw.rst

```diff
@@ -18,36 +19,110 @@ RGW Realm Operations
+
+The command supports providing the configuration through a spec file (`-i option`):
+
+.. prompt:: bash #
+
+  ceph rgw realm bootstrap -i myrgw.yaml
+
+Following is an example of RGW mutlisite spec file:
+
+.. code-block:: yaml
+
+  rgw_realm: myrealm
+  rgw_zonegroup: myzonegroup
+  rgw_zone: myzone
+  placement:
+    hosts:
+     - ceph-node-1
+     - ceph-node-2
+  spec:
+    rgw_frontend_port: 5500
+
+.. note:: The spec file used by RGW has the same format as the one used by the orchestrator. Thus,
+          the user can provide any orchestration supported rgw parameters including advanced
+          configuration features such as SSL certificates etc.
 
 
 Realm Credentials Token
 -----------------------
-A new token is created when bootstrapping a new realm, and also
-when creating one explicitly.  The token encapsulates
-the master zone endpoint, and a set of credentials that are associated
-with a system user.
-Removal of this token would remove the credentials, and if the corresponding
-system user has no more access keys, it is removed.
 
+Users can list the available tokens for the created (or already existing) realms.
+The token is a base64 string that encapsulates the realm information and its
+master zone endpoint authentication data. Following is an example of
+the `ceph rgw realm tokens` output:
+
+.. prompt:: bash #
+
+  ceph rgw realm tokens | jq
+
+.. code-block:: json
+
+  [
+    {
+      "realm": "myrealm1",
+      "token": "ewogICAgInJlYWxtX25hbWUiOiAibXlyZWFs....NHlBTFhoIgp9"
+    },
+    {
+      "realm": "myrealm2",
+      "token": "ewogICAgInJlYWxtX25hbWUiOiAibXlyZWFs....RUU12ZDB0Igp9"
+    }
+  ]
+
+User can use the token to create and synchronize a secondary zones
+on another cluster with the master zone by using `ceph rgw zone create`
+command and proving the corresponding token.
+
+Following is an example of zone spec file:
+
+.. code-block:: yaml
+
+  rgw_realm: myrealm
+  rgw_zonegroup: myzonegroup
+  rgw_zone: my-secondary-zone
+  rgw_realm_token: <token>
+  placement:
+    hosts:
+     - ceph-node-1
+     - ceph-node-2
+  spec:
+    rgw_frontend_port: 5500
+
+
+.. prompt:: bash #
+
+  ceph rgw zone create -i zone-spec.yaml
+
+.. note:: The spec file used by RGW has the same format as the one used by cephadm. Thus,
+          the user can provide any cephadm rgw supported parameter as any other advanced
+          configuration items such as SSL certificates etc.
 
 Commands
 --------
 ::
 
-  ceph rgw realm bootstrap
+  ceph rgw realm bootstrap -i spec.yaml
 
 Create a new realm + zonegroup + zone and deploy rgw daemons via the
-orchestrator.  Command returns a realm token that allows new zones to easily
-join this realm
+orchestrator using the information specified in the YAML file.
+
+::
+
+  ceph rgw realm tokens
+
+List the tokens of all the available realms
 
 ::
 
-  ceph rgw zone create
+  ceph rgw zone create -i spec.yaml
 
 Create a new zone and join existing realm (using the realm token)
```

Comment:
This could be `Join an existing realm by creating a new secondary zone (using the realm token)"

Reply:
fixed